src/common/enumerations.py

```diff
@@ -17,6 +17,17 @@
 
 from enum import Enum
 
+class StorageType(Enum):
+    """
+    Different types of underlying storage
+    """
+    FILE = 'file'
```

Comment:
FILE or object is a type
We should use FILESYSTEM and S3 as Storage Type. I would further prefer if we specialised it further as LOCAL_FILESYSTEM, PARALLEL_FILE_SYSTEM, and S3